licensing/README.md

```diff
@@ -1,13 +0,0 @@
-# Licensing
```

Comment:
Should we move this doc to the sig-arch directory in k/community - https://github.com/kubernetes/community/tree/master/sig-architecture?

Reply:
probably better nearer to where we point folks to when they update a dependency?
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/vendor.md

Reply:
> probably better nearer to where we point folks to when they update a dependency?

:+1:

Reply:
created https://github.com/kubernetes/community/pull/5400